test/gov/hzn_reg.sh

```diff
@@ -1,52 +1,147 @@
+cat <<EOF > /tmp/node_policy.json
+{
+  "properties": [
+    {
+      "name": "purpose",
+      "value": "network-testing"
+    },
+    {
+      "name": "group",
+      "value": "bluenode"
+    }
+  ],
+  "constraints": [
+    "iame2edev == true",
+    "NONS==false || NOGPS == false || NOLOC == false || NOPWS == false || NOHELLO == false"
+  ]
+}
+EOF
 
-echo -e "Define microservice using hzn dev:"
+cat <<EOF > /tmp/reg_userinput.json
+{
+  "services": [
+    {
+      "org": "IBM",
+      "url": "https://bluehorizon.network/services/netspeed",
+      "versionRange": "[2.2.0,INFINITY)",
+      "variables": {
+        "var1": "aString",
+        "var2": 5,
+        "var3": 22.2
+      }
+    },
+    {
+      "org": "e2edev@somecomp.com",
+      "url": "https://bluehorizon.network/services/netspeed",
+      "versionRange": "2.2.0",
+      "variables": {
+        "var1": "node_String",
+        "var2": 20,
+        "var3": 23.2
+      }
+    },
+    {
+      "org": "IBM",
+      "url": "https://bluehorizon.network/service-cpu",
+      "versionRange": "[0.0.0,INFINITY)",
+      "variables": {
+        "cpu_var1": "ibmnodevar1"
+      }
+    },
+    {
+      "org": "e2edev@somecomp.com",
+      "url": "https://bluehorizon.network/service-cpu",
+      "versionRange": "1.0.0",
+      "variables": {
+        "cpu_var1": "e2edevnodevar1"
+      }
+    },
+    {
+      "org": "e2edev@somecomp.com",
+      "url": "https://bluehorizon.network/services/locgps",
+      "versionRange": "2.0.3",
+      "variables": {
+        "test": "testValue",
+        "extra": "extraValue",
+        "HZN_LAT": 41.921766,
+        "HZN_LON": -73.894224,
+        "HZN_LOCATION_ACCURACY_KM": 0.5,
+        "HZN_USE_GPS": false
+      }
+    },
+    {
+      "org": "IBM",
+      "url": "https://bluehorizon.network/service-gps",
+      "versionRange": "2.0.3",
+      "variables": {
+        "HZN_LAT": 41.921766,
+        "HZN_LON": -73.894224,
+        "HZN_LOCATION_ACCURACY_KM": 0.5,
+        "HZN_USE_GPS": false
+      }
+    },
+    {
+      "org": "e2edev@somecomp.com",
+      "url": "https://bluehorizon.network/services/weather",
+      "versionRange": "1.5.0",
+      "variables": {
+        "HZN_LAT": 41.921766,
+        "HZN_LON": -73.894224,
+        "HZN_LOCATION_ACCURACY_KM": 0.5,
+        "HZN_USE_GPS": false,
+        "HZN_WUGNAME": "e2edev mocked pws",
+        "HZN_PWS_MODEL": "LaCrosse WS2317",
+        "MTN_PWS_MODEL": "LaCrosse WS2317",
+        "HZN_PWS_ST_TYPE": "WS23xx",
+        "MTN_PWS_ST_TYPE": "WS23xx"
+      }
+    }
+  ]
+}
+EOF
 
 
-cd /root/hzn/helloms
-hzn dev microservice publish -vI -k $KEY_TEST_DIR/*private.key
-if [ $? -ne 0 ]
-then
-    echo -e "hzn dev microservice publish failed."
-    exit 1
+# register
+if [ -n "$PATTERN" ]; then 
+	# register pattern sall
+	echo -e "  hzn register -u $E2EDEV_ADMIN_AUTH -n an12345:abcdefg -o e2edev@somecomp.com -f /tmp/reg_userinput.json -p e2edev@somecomp.com/$PATTERN"
```

Comment:
We need some additional tests that omit the -o flag, where the pattern is specified without the org, where the auth is specified without the org, where HZN_ORG_ID is used, etc etc. Also we need to test for the node already registered case, and the already unregistered case. Since it takes so long for all the agreements in the sall caseto be created, maybe we use the sns pattern (and the ns policy) to test this because that will have fewer workloads to start.